src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/cas3/Cas3BookingGapReportRepository.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas3
+
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+
+val QUERY = """
+select probation_region,
+  premises_id,
+  premises_name,
+  town,
+  bed_id,
+  bed_name,
+  year,
+  month,
+  days_booked,
+  (DATE_PART('days',
+    DATE_TRUNC('month', make_date(year, month, 1))
+      + '1 MONTH'::INTERVAL
+      - '1 DAY'::INTERVAL) - days_booked)::int as gap_days
+  from (select
+    probation_region,
+    premises_id,
+    premises_name,
+    town,
+    bed_id,
+    bed_name,
+    extract(year from booking_date)::int as year,
+    extract(month from booking_date)::int as month,
+    count(*) as days_booked
+    from (select
+      probation_regions.name as probation_region,
+      premises_id,
+      premises.name as premises_name,
+      town,
+      bed_id,
+      beds.name as bed_name,
+      generate_series(arrival_date, departure_date, interval '1 day') as booking_date
+      from bookings
+        left join premises on premises.id = bookings.premises_id
+        left join probation_regions on probation_regions.id = premises.probation_region_id
+        left join beds on beds.id = bookings.bed_id
+      where bookings.service = :service) q1
+      group by probation_region, premises_id, premises_name, town, bed_id, bed_name, year, month) q2
+order by probation_region, premises_name, year, month
+"""
+
+@Repository
+class Cas3BookingGapReportRepository(
+  val jdbcTemplate: NamedParameterJdbcTemplate,
+) {
+
+  fun generateBookingGapReport(service: String): String {
```

Comment:
Should this return a CSV object? Need to investigate.

Reply:
Currently return list of maps until we firm up requirements.

Reply:
OK for now since only required for the integration test at the moment.  The query can be run against the database directly until we confirm whether this is to be made into a proper report.